cannon/mipsevm/witness.go

```diff
@@ -6,8 +6,9 @@ type LocalContext common.Hash
 
 type StepWitness struct {
 	// encoded state witness
-	State     []byte
-	StateHash common.Hash
+	State         []byte
+	PreStateHash  common.Hash
+	PostStateHash common.Hash
```

Comment:
I feel like this is a little confusing - the hashes are explicitly pre / post, but `State` is ambiguous - is that a pre- or post- state?  Maybe `Step` should just return a pre-witness and a post-witness?

Reply:
Good point. I was trying to be less ambiguous calling `State` as `PreStateHash` but that had the opposite effect. And your suggestion is clearer.

I'd like to avoid generating additional proofs of memory if possible (for the post-state witness). So I've modified the cannon run command to generate the post-state hash by itself. Hopefully makes `StepWtiness` much less ambiguous.